integrations/cdi/oci-objectstorage-cdi/src/main/java/io/helidon/integrations/cdi/oci/objectstorage/OciConfigConfigSource.java

```diff
@@ -98,32 +100,45 @@ public int getOrdinal() {
   @Override
   public String getValue(final String propertyName) {
     final String returnValue;
-    if (propertyName == null || propertyName.equals("oci.auth.profile") || propertyName.equals("oci.config.path")) {
+    if (propertyName == null) {
       returnValue = null;
+    } else if (propertyName.equals(ConfigSource.CONFIG_ORDINAL)) {
+      returnValue = String.valueOf(this.getOrdinal());
     } else {
       Map<String, String> properties = this.properties;
       if (properties == null) {
-        final Config config = ConfigProvider.getConfig();
-        assert config != null;
-        final String profile = config.getOptionalValue("oci.auth.profile", String.class).orElse("DEFAULT");
-        assert profile != null;
-        final String configFilePath = config.getOptionalValue("oci.config.path", String.class).orElse(null);
-        final ConfigFileAuthenticationDetailsProvider provider;
-        ConfigFileAuthenticationDetailsProvider temp = null;
+        properties = PROPERTIES.get();
+      }
+      if (properties == null) {
+        PROPERTIES.set(Collections.emptyMap());
         try {
-          if (configFilePath == null) {
-            temp = new ConfigFileAuthenticationDetailsProvider(profile);
-          } else {
-            temp = new ConfigFileAuthenticationDetailsProvider(configFilePath, profile);
+          // Reentrant call; i.e. this causes a new
+          // OciConfigConfigSource instance to be created and queried
+          // for various values.
+          final Config config = ConfigProvider.getConfig();
+          assert config != null;
```

Comment:
Please do not use `assert` in the code. I do not see an advantage checking for null not being returned by methods that claim to never return null (such as `Optional.orElse(T)`).
We are not using this approach anywhere in Helidon.
No need to remove existing `assert` statements from code, but please do not add new ones.